app/services/importer/greenhouse_fields_formatter.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+# This service object is responsible for formatting Greenhouse data into a structured format suitable for our application.
+# It extracts specific sections (core questions, demographics, compliance, location) from the raw data and transforms them into a standardized format.
+# The output of this service is used by greenhouse_fields_builder.
+
+# Key functionalities:
+# - Extracts core questions directly from the raw data.
+# - Formats demographic and compliance questions with title, description, and transformed questions.
+# - Creates a location question with auto-complete input for city.
+module Importer
+  class GreenhouseFieldsFormatter < ApplicationTask
+    def initialize(data)
+      @data = data
+    end
+
+    def call
+      return unless processable
+
+      process
+    end
+
+    private
+
+    def processable
+      @data
+    end
+
+    def process
+      select_transform_data
+    end
+
+    def select_transform_data
+      {
+        core_questions: { title: nil, description: nil, questions: @data[:questions] },
+        demographic_questions: demographic_formatter(@data.dig(:demographic_questions, :questions)),
+        compliance_questions: compliance_formatter(@data[:compliance]),
+        location_questions: { title: nil, description: nil, questions: location_formatter(@data[:location_questions]) }
+      }
+    end
+
+    ###
+
+    def compliance_formatter(compliance_data)
+      return {} unless compliance_data
+
+      {
+        title: section_attribute_finder(:compliance, :title) || "EEOC compliance questions",
+        description: section_attribute_finder(:compliance, :description),
+        questions: compliance_questions(compliance_data)
+      }
+    end
+
+    def compliance_questions(compliance_data)
+      compliance_data.map do |section|
```

Comment:
Interesting that we have all these nested maps - do we need to be conscious of how this performs?